### DISCUSSION TITLE
Just confirmed that the binary in the Docker image (bookworm) has much lower performance than Github Releases

### DISCUSSION THREAD

**Discussion by bohanwood** [php/frankenphp#802] [General]
- [Docker image](https://hub.docker.com/layers/dunglas/frankenphp/1-php8.3/images/sha256-8d89d357e46a896fed24f2518a7146e731db0474872af3f76d29981e29c73a6f?context=explore)
- [GitHub Release](https://github.com/dunglas/frankenphp/releases/download/v1.1.5/frankenphp-linux-x86_64)

Machine:
- i5-12500H (16C)
- 16G RAM
- WSL2

It's a Symfony project, using the symfony-docker template.

The interesting thing:
When using MySQL there's not much difference but when using SQLite as DB,
`dunglas/frankenphp` can only have 2000-3000 QPS while the GitHub one can go 5000-7000 QPS

Sorry for can't provide reproducible environment for now

**Comment by dunglas** [comment]
Have you checked that the file containing the SQLite DB isn't in a volume or something like that?

**Reply by withinboredom** [reply]
I do believe you want the sqlite db running in a volume, otherwise it is going to be participating in the COW filesystem, which is likely quite slow.

**Reply by bohanwood** [reply]
I did this in a docker compose. Both container uses the DB in a named volume.

**Reply by bohanwood** [reply]
The biggest difference is that they use different PHP:
1. Container A: `FROM php:8.3` and downloaded the static `frankenphp` binary
2. Container B: `FROM dunglas/frankenphp:1-php8.3`

Same SQLite DB in same named volume. A has better QPS than B.

When checking phpinfo,
 - **A** did not load php.ini from `/usr/local/etc/php`. And the PDO SQLite version is 3.43.2.
 - **B** loaded php.ini from `/usr/local/etc/php` and extensions from `install-php-extensions` in the Dockerfile. And the version is 3.40.1.
 
 So basically they use completely different PHP environment. However this still didn't show what leds to the performance difference.
 
 Any ideas?

**Reply by withinboredom** [reply]
Musl vs. libc vs. static compilation vs. shared dependencies? It could also be that the host system was under different priorities and loads at the time you tested ... or even the order can affect it (cold vs. hot kernel file caches). There can also be differences in workloads (for example, I see the exact opposite of what you see on my test workloads).

There are simply too many variables for there to be a definitive "this is better than that."
